sdk/provisioning/Azure.Provisioning.CloudMachine/api/Azure.Provisioning.CloudMachine.netstandard2.0.cs

```diff
@@ -46,30 +44,34 @@ namespace Azure.Core
     public partial class ClientCache
     {
         public ClientCache() { }
-        public T Get<T>(string id, System.Func<T> value) where T : class { throw null; }
+        public T Get<T>(System.Func<T> value, string? id = null) where T : class { throw null; }
+    }
+    public enum ClientConnectionKind
+    {
+        EntraId = 0,
+        ApiKey = 1,
+        OutOfBand = 2,
     }
     [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
-    public readonly partial struct ClientConfiguration
+    public readonly partial struct ClientConnectionOptions
     {
         private readonly object _dummy;
         private readonly int _dummyPrimitive;
-        public ClientConfiguration(string endpoint, string? apiKey = null) { throw null; }
-        public string? ApiKey { get { throw null; } }
-        public Azure.Core.CredentialType CredentialType { get { throw null; } }
-        public string Endpoint { get { throw null; } }
+        public ClientConnectionOptions(string subclientId) { throw null; }
+        public ClientConnectionOptions(System.Uri endpoint, Azure.Core.TokenCredential credential) { throw null; }
+        public ClientConnectionOptions(System.Uri endpoint, string apiKey) { throw null; }
+        public string? ApiKeyCredential { get { throw null; } }
+        public Azure.Core.ClientConnectionKind ConnectionKind { get { throw null; } }
+        public System.Uri? Endpoint { get { throw null; } }
+        public string? Id { get { throw null; } }
+        public Azure.Core.TokenCredential? TokenCredential { get { throw null; } }
     }
-    public enum CredentialType
+    public abstract partial class ClientWorkspace
```

Comment:
Nit - consider calling this a `ClientFabric` if you want to signal how cool your project is.  :smile:

Reply:
Yeah, we need to decide what to call this concept. Azure.AI project has that. I am basically prototyping what I would like to do in Azure.AI